### PR TITLE
feat(bigquery): add reservation usage stats to query statistics

### PR DESCRIPTION
**Pull request by shollyman**
This PR plumbs through reservation usage stats reported per-job.

Also added a small integration test for testing query statistics,
as we didn't have one previously.

### CODE REVIEW

File: bigquery/integration_test.go [hunk 1]
```diff
@@ -1630,6 +1630,63 @@ func TestIntegration_TableUpdate(t *testing.T) {
...
+	if qStats.CacheHit {
+		t.Error("unexpected cache hit")
+	}
+
+	if qStats.StatementType != "SELECT" {
+		t.Errorf("expected SELECT statement type, got: %s", qStats.StatementType)
+	}
+
+	if len(qStats.QueryPlan) == 0 {
+		t.Errorf("expected query plan, none present")
```
Comment (codyoss) [inline bigquery/integration_test.go:1682]: nit: t.Error

File: bigquery/integration_test.go [hunk 2]
```diff
@@ -1630,6 +1630,63 @@ func TestIntegration_TableUpdate(t *testing.T) {
...
+	}
+
+	if len(qStats.Timeline) == 0 {
+		t.Errorf("expected query timeline, none present")
```
Comment (codyoss) [inline bigquery/integration_test.go:1686]: nit: t.Error

File: bigquery/job.go [hunk 3]
```diff
@@ -439,6 +439,9 @@ type QueryStatistics struct {
 
 	// The DDL target table, present only for CREATE/DROP FUNCTION/PROCEDURE queries.
 	DDLTargetRoutine *Routine
+
+	// Information about reservation usage.
+	ReservationUsage []*ReservationUsage
```
Comment (codyoss) [inline bigquery/job.go:444]: nit: "// ReservationUsage ..."

File: bigquery/job.go [hunk 4]
```diff
@@ -561,6 +564,25 @@ type QueryTimelineSample struct {
 	SlotMillis int64
 }
 
+// ReservationUsage contains information about a job's usage of a single reservation.
+type ReservationUsage struct {
+	// Slot-milliseconds the job spent in the given reservation.
+	SlotMillis int64
```
Comment (codyoss) [inline bigquery/job.go:570]: nit: "// SlotMillis ..."

**Review by codyoss — APPROVED** [review]
LGTM, just a few nits

**Comment by shollyman** [comment]
Just did a secondary review.  This is a mirrored stat, hoisting this up outside of query stats.

**Pull request merged**